### PR TITLE
feat(Salad): add presence

### PR DESCRIPTION
**Pull request by daglaroglou**
## Acknowledgements
- [ ] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [ ] I linted the code by running `yarn format`
- [ ] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/S/Salad/metadata.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.9",
```
Comment (DarkVillager) [inline websites/S/Salad/metadata.json:2]:
```suggestion
	"$schema": "https://schemas.premid.app/metadata/1.10",
```

File: websites/S/Salad/presence.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const enum Assets {
+	Logo = "https://cdn.solo.to/user/a/63654eae75b1e7_99721437.gif",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		startTimestamp: browsingTimestamp,
+	};
+	const privacy = await presence.getSetting<boolean>("privacy");
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:16]:
```suggestion
	},
	privacy = await presence.getSetting<boolean>("privacy"),
	{ hostname, pathname } = document.location;
```

File: websites/S/Salad/presence.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+	if (document.location.pathname.includes("/earn/summary")) {
+		presenceData.details = "In Summary";
+		presenceData.state = "Balance: " + document.querySelector(".c0116").textContent;
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:20]:
```suggestion
		presenceData.state = `Balance: ${document.querySelector(".c0116").textContent}`
```

File: websites/S/Salad/presence.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		if (privacy) {
+			presenceData.state = "";
+		}
+	}
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:24]:
```suggestion
	}
```

File: websites/S/Salad/presence.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+		presenceData.details = "In Community";
+		presenceData.state = "Browsing...";
+		presenceData.smallImageKey = Assets.Search;
+		if (document.location.pathname.includes("-")) {
+			presenceData.details = "In Community";
+			presenceData.state = document.title.split("- Salad Chef Community")[0].trim();
+			presenceData.smallImageKey = Assets.Reading;
+		}
+	}
+	presence.setActivity(presenceData);
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:99]:
```suggestion
if (privacy && presenceData.state) delete presenceData.state
	presence.setActivity(presenceData);
```

File: websites/S/Salad/presence.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		startTimestamp: browsingTimestamp,
+	};
+	const privacy = await presence.getSetting<boolean>("privacy");
+
+	if (document.location.pathname.includes("/earn/summary")) {
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:18]: Simplify it, do this for all mentions of this.
```suggestion
	if (pathname.includes("/earn/summary")) {
```

File: websites/S/Salad/presence.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+			presenceData.state = document.title.split("- Salad Support")[0].trim();
+			presenceData.smallImageKey = Assets.Reading;
+		}
+		else if (document.location.pathname.includes("category")) {
+			presenceData.details = "In Article Categories";
+			presenceData.state = document.title.split("- Salad Support")[0].trim();
+			presenceData.smallImageKey = Assets.Search;
+		}
+	}
+	else if (document.location.hostname.startsWith("community")) {
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:89]:
```suggestion
	else if (hostname.startsWith("community")) {
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
Just some changes. If you have questions feel free to ask on discord in the presence-dev channel or in my dms: @dark_ville

**Comment by DarkVillager** [comment]
Also change the title to: feat(Salad): add presence

File: websites/S/Salad/metadata.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+		"id": "852825042630475827",
+		"name": "daglaroglou"
+	},
+	"service": "Salad",
+	"description": {
+		"en": "Salad is a worldwide computesharing community composed of thousand of gamers. Our users, the mighty Salad Chefs, share resources from their idle PCs, power advanced computing projects, and score real-life-rewards."
+	},
+	"url": ["salad.com", "community.salad.com", "support.salad.io"],
+	"version": "1.0.0",
+	"logo": "https://blog.salad.com/wp-content/uploads/2023/08/Salad-Logo.png",
```
Comment (theusaf) [inline websites/S/Salad/metadata.json:13]: Logo should be 512x512px

File: websites/S/Salad/presence.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+		presenceData.details = "In Store";
+		presenceData.state = "Browisng...";
+		presenceData.smallImageKey = Assets.Search;
+	}
+	else if (document.location.pathname.endsWith("/download")) {
+		presenceData.details = "Downloading Salad";
+	}
+	else if (document.location.pathname.includes("/search")) {
+		presenceData.details = "In Store";
+		presenceData.state = "Searching for " + (document.documentURI.split("?q=")[1]).split("&")[0].trim();
```
Comment (theusaf) [inline websites/S/Salad/presence.ts:52]: Use `document.location.href` instead of `documentURI`

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by daglaroglou** [comment]
can you change that please? ill be away from home for some days...

**Comment by daglaroglou** [comment]
updated stuff

File: websites/S/Salad/presence.ts [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+		presenceData.smallImageKey = Assets.Search;
+		if (document.location.pathname.includes("-")) {
+			presenceData.details = "In Community";
+			presenceData.state = document.title.split("- Salad Chef Community")[0].trim();
+			presenceData.smallImageKey = Assets.Reading;
+		}
+	}
+if (privacy && presenceData.state) delete presenceData.state
+	presence.setActivity(presenceData);
+});
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:99]:
```suggestion
const presence = new Presence({
		clientId: "911546577485725706",
	}),
	browsingTimestamp = Math.floor(Date.now() / 1000);

const enum Assets {
	Logo = "https://i.imgur.com/Lc9wcdl.gif",
}

presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
			largeImageKey: Assets.Logo,
			startTimestamp: browsingTimestamp,
		},
		privacy = await presence.getSetting<boolean>("privacy"),
		{ hostname, pathname } = document.location;

	if (hostname.startsWith("support")) {
		if (pathname.includes("troubleshooting"))
			presenceData.state = "Troubleshooting";
		else if (pathname.includes("app-guides")) presenceData.state = "App guides";
		else if (pathname.includes("faq")) presenceData.state = "The faq";
		else if (pathname.includes("company")) presenceData.state = "Company";
		else if (pathname.includes("article")) {
			presenceData.details = privacy
				? "Support - reading an article"
				: "Support - Reading article";
			presenceData.state = document.querySelector(".title")?.textContent;
			presenceData.smallImageKey = Assets.Reading;
		} else if (pathname.includes("category")) {
			presenceData.details = privacy
				? "Support - viewing an article category"
				: "Support - viewing article category";
			presenceData.state = document.querySelector(".title")?.textContent;
			presenceData.smallImageKey = Assets.Search;
		} else presenceData.state = "Home";

		if (!presenceData.details)
			presenceData.details = privacy
				? "Support - viewing"
				: "Support - browsing";
	} else if (hostname.startsWith("community")) {
		if (pathname.includes("-")) {
			presenceData.details = privacy
				? "Community - reading a blog post"
				: "Community - reading about";
			presenceData.state = document.querySelector(".xpro-post-title");
			presenceData.smallImageKey = Assets.Reading;
		} else {
			presenceData.details = "Community - Browsing...";
			presenceData.smallImageKey = Assets.Search;
		}
	} else if (pathname.includes("/earn/summary")) {
		presenceData.details = privacy
			? "Viewing their summary"
			: "Summary - earnings";
		presenceData.state = `Balance: ${
			document.querySelector(".c0116")?.textContent
		}`;
	} else if (pathname.includes("/account/")) {
		presenceData.details = privacy
			? "Viewing their account settings"
			: "Account - viewing";
		if (pathname.endsWith("/summary"))
			presenceData.state = "Account management";
		else if (pathname.endsWith("/referrals")) presenceData.state = "Referrals";
		else if (pathname.endsWith("/bonuses")) presenceData.state = "Bonuses";
	} else if (pathname.includes("/rewards/")) {
		presenceData.details = privacy
			? "Store - viewing all items"
			: "Store - Viewing";
		presenceData.state = document?.title?.split("|")?.[0];
		presenceData.smallImageKey = Assets.Viewing;
	} else if (pathname.endsWith("/store")) {
		presenceData.details = "Store - browsing";
		presenceData.smallImageKey = Assets.Search;
	} else if (pathname.includes("/search")) {
		presenceData.details = privacy
			? "Store - searching for something"
			: "Store - searching for";
		presenceData.state = `Searching for ${
			document.querySelector<HTMLInputElement>('[placeholder="SEARCH"]')?.value
		}`;
		presenceData.smallImageKey = Assets.Search;
	} else if (pathname.endsWith("/download"))
		presenceData.details = "Viewing the download page";
	else if (pathname.endsWith("/vault"))
		presenceData.details = "Vault - viewing their vault";

	if (privacy && presenceData.state) delete presenceData.state;
	presence.setActivity(presenceData);
});

```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
I love your attempt, but the code is too low quality. But since you put in effort, i'll suggest this change, that will make the presence high enough quality for the store.

If you have any questions, dm me on discord (@Dark_Ville) 

Changes include:
* Logo should be 512x512
* State & Details can only start with a capital (just like a normal sentence)
* Remove duplicate code
* Sort ifs to be more readable
* [Destructure variables ](https://dmitripavlutin.com/javascript-object-destructuring/)
* Change document.title to selectors so we don't have to split
* Implement privacy in a way so you have logical structure

File: websites/S/Salad/presence.ts [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+				? "Support - viewing an article category"
+				: "Support - viewing article category";
+			presenceData.state = document.querySelector(".title")?.textContent;
+			presenceData.smallImageKey = Assets.Search;
+		} else presenceData.state = "Home";
+
+		if (!presenceData.details)
+			presenceData.details = privacy
+				? "Support - viewing"
+				: "Support - browsing";
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:41]:
```suggestion
		if (!presenceData.details) {
			presenceData.details = privacy
				? "Support - viewing"
				: "Support - browsing";
			}
```

File: websites/S/Salad/presence.ts [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+		presenceData.smallImageKey = Assets.Search;
+	} else if (pathname.endsWith("/download"))
+		presenceData.details = "Viewing the download page";
+	else if (pathname.endsWith("/vault"))
+		presenceData.details = "Vault - viewing their vault";
+
+	if (privacy && presenceData.state) delete presenceData.state;
+	presence.setActivity(presenceData);
+});
+
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:93]:
```suggestion
```

**Review by DarkVillager — APPROVED** [review]
I forgot to remove this and fix that, other than that LGTM

File: websites/S/Salad/presence.ts [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+		}`;
+		presenceData.smallImageKey = Assets.Search;
+	} else if (pathname.endsWith("/download"))
+		presenceData.details = "Viewing the download page";
+	else if (pathname.endsWith("/vault"))
+		presenceData.details = "Vault - viewing their vault";
+
+	if (privacy && presenceData.state) delete presenceData.state;
+	presence.setActivity(presenceData);
+});
```
Comment (DarkVillager) [inline websites/S/Salad/presence.ts:93]:
```suggestion
const presence = new Presence({
		clientId: "911546577485725706",
	}),
	browsingTimestamp = Math.floor(Date.now() / 1000);

const enum Assets {
	Logo = "https://i.imgur.com/Lc9wcdl.gif",
}

presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
			largeImageKey: Assets.Logo,
			startTimestamp: browsingTimestamp,
		},
		privacy = await presence.getSetting<boolean>("privacy"),
		{ hostname, pathname } = document.location;

	if (hostname.startsWith("support")) {
		if (pathname.includes("troubleshooting"))
			presenceData.state = "Troubleshooting";
		else if (pathname.includes("app-guides")) presenceData.state = "App guides";
		else if (pathname.includes("faq")) presenceData.state = "The faq";
		else if (pathname.includes("company")) presenceData.state = "Company";
		else if (pathname.includes("article")) {
			presenceData.details = privacy
				? "Support - reading an article"
				: "Support - Reading article";
			presenceData.state = document.querySelector(".title")?.textContent;
			presenceData.smallImageKey = Assets.Reading;
		} else if (pathname.includes("category")) {
			presenceData.details = privacy
				? "Support - viewing an article category"
				: "Support - viewing article category";
			presenceData.state = document.querySelector(".title")?.textContent;
			presenceData.smallImageKey = Assets.Search;
		} else presenceData.state = "Home";

		if (!presenceData.details) {
			presenceData.details = privacy
				? "Support - viewing"
				: "Support - browsing";
		}
	} else if (hostname.startsWith("community")) {
		if (pathname.includes("-")) {
			presenceData.details = privacy
				? "Community - reading a blog post"
				: "Community - reading about";
			presenceData.state =
				document.querySelector(".xpro-post-title")?.textContent;
			presenceData.smallImageKey = Assets.Reading;
		} else {
			presenceData.details = "Community - Browsing...";
			presenceData.smallImageKey = Assets.Search;
		}
	} else if (pathname.includes("/earn/summary")) {
		presenceData.details = privacy
			? "Viewing their summary"
			: "Summary - earnings";
		presenceData.state = `Balance: ${
			document.querySelector(".c0116")?.textContent
		}`;
	} else if (pathname.includes("/account/")) {
		presenceData.details = privacy
			? "Viewing their account settings"
			: "Account - viewing";
		if (pathname.endsWith("/summary"))
			presenceData.state = "Account management";
		else if (pathname.endsWith("/referrals")) presenceData.state = "Referrals";
		else if (pathname.endsWith("/bonuses")) presenceData.state = "Bonuses";
	} else if (pathname.includes("/rewards/")) {
		presenceData.details = privacy
			? "Store - viewing all items"
			: "Store - Viewing";
		presenceData.state = document?.title?.split("|")?.[0];
		presenceData.smallImageKey = Assets.Viewing;
	} else if (pathname.endsWith("/store")) {
		presenceData.details = "Store - browsing";
		presenceData.smallImageKey = Assets.Search;
	} else if (pathname.includes("/search")) {
		presenceData.details = privacy
			? "Store - searching for something"
			: "Store - searching for";
		presenceData.state = `Searching for ${
			document.querySelector<HTMLInputElement>('[placeholder="SEARCH"]')?.value
		}`;
		presenceData.smallImageKey = Assets.Search;
	} else if (pathname.endsWith("/download"))
		presenceData.details = "Viewing the download page";
	else if (pathname.endsWith("/vault"))
		presenceData.details = "Vault - viewing their vault";

	if (privacy && presenceData.state) delete presenceData.state;
	presence.setActivity(presenceData);
});
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
format

**Review by DarkVillager — APPROVED** [review]
LGTM

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**